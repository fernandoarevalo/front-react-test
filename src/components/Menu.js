import React, { useState } from "react";
import styled from "styled-components";
import MenuItem from "components/MenuItem";
// Utils
import AIRLINES from "utils/mockups/Airlines";
import { MENU } from "utils/constants/Contants";

const MenuContainer = styled.ul`
  display: flex;
  text-align: center;
  margin-top: 0;

  li {
    flex: 1 1 auto;
  }

  li:active {
    background-color: initial;
  }

  @media screen and (max-width: 920px) {
    display: block;
  }
`;

export default function Menu() {
  const [airLine, setAirLine] = useState(AIRLINES[0]);

  return (
    <React.Fragment key="menu">
      <MenuContainer className="card collection" data-testid="menu-container">
        {AIRLINES.map((airlineObj) => {
          return (
            <MenuItem
              key={airlineObj.id}
              airlineObj={airlineObj}
              selected={airLine === airlineObj}
              setAirLine={setAirLine}
            />
          );
        })}
      </MenuContainer>
      <section className="container">
        <div className="row">
          <div className="col s12">
            <h4>{`${MENU.TITLE} ${airLine?.name}`}</h4>
          </div>
        </div>
      </section>
    </React.Fragment>
  );
}
