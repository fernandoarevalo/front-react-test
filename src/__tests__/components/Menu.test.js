import React from "react";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import Menu from "components/Menu";

describe("<Menu />", () => {
  afterEach(() => {
    cleanup();
  });

  test("Menu container section", () => {
    const { getByTestId } = render(<Menu />);
    const container = getByTestId("menu-container");
    expect(container).toBeTruthy();
  });

  test("Menu container section", () => {
    const { getByTestId } = render(<Menu />);
    const container = getByTestId("menu-container");
    expect(container).toBeTruthy();
  });

  test("Snapshot Menu component", () => {
    const { asFragment } = render(<Menu />);
    expect(asFragment()).toMatchSnapshot();
  });
});
