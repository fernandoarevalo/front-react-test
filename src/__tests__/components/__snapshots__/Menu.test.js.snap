// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Snapshot Menu component 1`] = `
<DocumentFragment>
  <ul
    class="sc-gsTCUz lgfLoM card collection"
    data-testid="menu-container"
  >
    <li
      aria-selected="true"
      class="collection-item active"
      role="tab"
    >
      <button
        class="sc-bdfBwQ hGbOiT"
        type="button"
      >
        Vivaair
      </button>
    </li>
    <li
      aria-selected="false"
      class="collection-item "
      role="tab"
    >
      <button
        class="sc-bdfBwQ hGbOiT"
        type="button"
      >
        Avianca
      </button>
    </li>
  </ul>
  <section
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col s12"
      >
        <h4>
          Hola, bienvenido, sabemos que quieres viajar en un Vivaair
        </h4>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
